api/v1beta1/watcherapplier_types.go

```diff
@@ -28,13 +29,34 @@ type WatcherApplierSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	WatcherCommon `json:",inline"`
+
+	// +kubebuilder:validation:Required
+	// Secret containing all passwords / keys needed
+	Secret string `json:"secret"`
+
 	WatcherSubCrsCommon `json:",inline"`
 }
 
 // WatcherApplierStatus defines the observed state of WatcherApplier
 type WatcherApplierStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	// Conditions
+	Conditions condition.Conditions `json:"conditions,omitempty" optional:"true"`
+
+	// ObservedGeneration - the most recent generation observed for this
+	// service. If the observed generation is less than the spec generation,
+	// then the controller has not processed the latest changes injected by
+	// the openstack-operator in the top-level CR (e.g. the ContainerImage)
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// ReadyCount of watcher API instances
```

Comment:
I wouldn't add ReadyCount until we create the deployment and can get the value.

Reply:
Yes, make sense, I commented out and add a TODO to add it back afterwards.